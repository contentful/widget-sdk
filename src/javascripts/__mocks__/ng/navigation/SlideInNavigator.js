export const goToSlideInEntity = jest.fn();
export const goToPreviousSlideOrExit = jest.fn();

export default {
  goToSlideInEntity,
  goToPreviousSlideOrExit
};
