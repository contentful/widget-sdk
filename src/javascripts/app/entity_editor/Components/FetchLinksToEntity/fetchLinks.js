import { makeEntityRef, href } from 'states/Navigator';
import { EntityType } from '../constants';
import TheLocaleStore from 'services/localeStore';
import { getModule } from 'NgRegistry';
import * as EntityHelpers from 'app/entity_editor/entityHelpers';
import { onFetchLinks } from 'analytics/events/IncomingLinks';

const assertEntityTypeIsValid = entityType => {
  if (entityType !== EntityType.ENTRY && entityType !== EntityType.ASSET) {
    throw new Error(`Unsupported entityType ${entityType}`);
  }
};

export default (id, type) => {
  const spaceContext = getModule('spaceContext');

  const defaultLocaleCode = TheLocaleStore.getDefaultLocale().code;
  const entityHelpers = EntityHelpers.newForLocale(defaultLocaleCode);

  assertEntityTypeIsValid(type);
  const payloadKey = type === EntityType.ENTRY ? 'links_to_entry' : 'links_to_asset';
  const payload = {
    [payloadKey]: id
  };

  return spaceContext.cma.getEntries(payload).then(({ items }) => {
    onFetchLinks({
      entityId: id,
      entityType: type,
      incomingLinksCount: items.length
    });

    return Promise.all(
      items.map(entry => {
        const { id } = entry.sys;
        const contentType = spaceContext.publishedCTs.get(entry.sys.contentType.sys.id);
        // Remove query params of the current URL. This prevents
        // opening slide-in editor when navigating to incoming links.
        return entityHelpers.entityTitle(entry).then(title => ({
          id,
          title,
          contentTypeName: contentType.data.name,
          url: href(makeEntityRef(entry))
            .split('?')
            .shift()
        }));
      })
    );
  });
};
