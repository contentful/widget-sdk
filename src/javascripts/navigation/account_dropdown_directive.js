'use strict';

angular.module('contentful')

.directive('cfAccountDropdown', ['require', function (require) {
  return {
    template: require('navigation/templates/AccountDropdown.template').default(),
    restrict: 'E',
    scope: {user: '='},
    controller: 'cfAccountDropdownController'
  };
}])

.controller('cfAccountDropdownController', ['$scope', 'require', function ($scope, require) {
  var TheAccountView = require('TheAccountView');
  var authentication = require('Authentication');
  var K = require('utils/kefir');
  var Config = require('Config');
  var analytics = require('analytics/Analytics');
  var intercom = require('intercom');

  K.onValueScope($scope, TheAccountView.canShowIntercomLink$, function (canShowIntercomLink) {
    $scope.canShowIntercomLink = canShowIntercomLink;
  });

  $scope.userProfileRef = {
    path: ['account', 'profile', 'user'],
    options: { reload: true }
  };

  $scope.supportUrl = Config.supportUrl;
  $scope.openIntercom = intercom.open;
  $scope.isIntercomLoaded = intercom.isLoaded;
  $scope.logout = logout;

  function logout () {
    analytics.track('global:logout_clicked');
    authentication.logout();
  }
}]);
