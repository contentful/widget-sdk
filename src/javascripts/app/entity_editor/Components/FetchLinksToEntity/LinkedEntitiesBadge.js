import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  Icon,
  Dropdown,
  List,
  ListItem,
  TextLink,
  Paragraph
} from '@contentful/forma-36-react-components';
import * as slideInNavigator from 'navigation/SlideInNavigator';
import tokens from '@contentful/forma-36-tokens';
import { css, cx } from 'emotion';
import FetchLinksToEntity from './index';

const styles = {
  badge: css({
    display: 'inline-flex',
    alignItems: 'center',
    justifyContent: 'center',
    background: '#8C53C2',
    fontSize: tokens.fontSizeL,
    padding: '0px 5px',
    borderRadius: '4px',
    color: tokens.colorWhite
  }),
  idle: css({
    background: tokens.colorElementLight,
    color: tokens.colorTextMid
  }),
  linksLength: css({
    color: 'inherit'
  }),
  icon: css({
    marginRight: '2px'
  }),
  dropdown: css({
    lineHeight: 0
  }),
  linkList: css({
    padding: 0,
    maxHeight: '200px',
    overflowY: 'auto',
    overflowX: 'hidden'
  }),
  linkListHeader: css({
    padding: `0 ${tokens.spacingXs}`,
    background: tokens.colorElementLight
  }),
  entryLinkWrapper: css({
    cursor: 'pointer',
    display: 'flex',
    padding: tokens.spacingXs,
    '&:hover': {
      background: tokens.colorElementLightest
    }
  }),
  entryLink: css({
    marginRight: tokens.spacingL,
    maxWidth: '400px',
    color: tokens.colorTextDark,
    textDecoration: 'none',
    span: {
      overflow: 'hidden',
      whiteSpace: 'nowrap',
      textOverflow: 'ellipsis'
    }
  }),
  contentType: css({
    marginLeft: 'auto'
  })
};

export default function LinkedEntitiesBadge({ entityInfo, className }) {
  const [isLinkListOpen, setLinkListOpen] = useState(false);
  const [dropdownContainer, setDropdownContainer] = useState(null);
  return (
    <FetchLinksToEntity
      {...entityInfo}
      render={({ links }) => {
        const title = `${links.length === 1 ? 'Used' : 'Reused'} by ${links.length} ${
          links.length === 1 ? 'entry' : 'entries'
        }`;

        const isReused = links.length > 1;

        return (
          <Dropdown
            isOpen={isLinkListOpen}
            className={styles.dropdown}
            getContainerRef={ref => {
              setDropdownContainer(ref);
            }}
            toggleElement={
              <div
                data-test-id="cf-linked-entities-icon"
                title={title}
                className={cx(styles.badge, (!links || !isReused) && styles.idle, className)}
                onMouseEnter={() => links.length > 0 && setLinkListOpen(true)}
                onMouseLeave={e => {
                  if (
                    e.relatedTarget === window ||
                    !dropdownContainer.contains(e.relatedTarget) ||
                    !dropdownContainer === e.relatedTarget
                  ) {
                    setLinkListOpen(false);
                  }
                }}>
                <Icon icon="Link" color={isReused ? 'white' : 'muted'} className={styles.icon} />
                <Paragraph className={styles.linksLength}>{links.length}</Paragraph>
              </div>
            }>
            <div
              onMouseEnter={() => setLinkListOpen(true)}
              onMouseLeave={() => setLinkListOpen(false)}>
              <div className={styles.linkListHeader} data-test-id="cf-linked-entities-header">
                <Paragraph>{title}</Paragraph>
              </div>
              <List className={styles.linkList}>
                {links.map(link => (
                  <ListItem
                    key={link.id}
                    testId={`cf-linked-entry-${link.id}`}
                    className={styles.entryLinkWrapper}
                    onClick={e => {
                      e.stopPropagation();
                      setLinkListOpen(false);
                      slideInNavigator.goToSlideInEntity({
                        type: 'Entry',
                        id: link.id
                      });
                    }}>
                    <TextLink className={styles.entryLink}>{link.title || 'Untitled'}</TextLink>
                    <Paragraph className={styles.contentType}>{link.contentTypeName}</Paragraph>
                  </ListItem>
                ))}
              </List>
            </div>
          </Dropdown>
        );
      }}
    />
  );
}

LinkedEntitiesBadge.propTypes = {
  entityInfo: PropTypes.shape({
    id: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired
  }),
  className: PropTypes.string
};
