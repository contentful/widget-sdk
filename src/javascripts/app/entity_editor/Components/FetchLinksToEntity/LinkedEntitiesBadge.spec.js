import React from 'react';
import { render, wait, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import LinkedEntitiesBadge from './LinkedEntitiesBadge';
import { Promise } from 'bluebird';
import * as slideInNavigator from 'navigation/SlideInNavigator';

import fetchLinks from 'app/entity_editor/Components/FetchLinksToEntity/fetchLinks';

jest.mock('navigation/SlideInNavigator', () => ({
  goToSlideInEntity: jest.fn()
}));

jest.mock('app/entity_editor/Components/FetchLinksToEntity/fetchLinks', () => jest.fn());

jest.mock('services/localeStore', () => ({
  getDefaultLocale: () => ({
    code: 'en-US'
  }),
  toInternalCode: () => jest.fn()
}));

const mockedLinks = [
  {
    id: 'id1',
    title: 'title1',
    contentTypeName: 'contentTypeName1',
    url: 'url1'
  },
  {
    id: 'id2',
    title: 'title2',
    contentTypeName: 'contentTypeName2',
    url: 'url2'
  }
];

describe('LinkedEntitiesBadge', () => {
  const build = props => {
    const resultProps = {
      entityInfo: {
        id: 'entityId',
        type: 'Entry'
      }
    };

    return [render(<LinkedEntitiesBadge {...resultProps} />), props];
  };

  it('renders the component with 2 linked entries', async () => {
    fetchLinks.mockImplementation(() => Promise.resolve(mockedLinks));
    const [renderResult] = build();
    await wait();
    fireEvent.mouseOver(renderResult.getByTestId('cf-linked-entities-icon'));
    expect(renderResult).toMatchSnapshot();
  });

  it('does not render with 1 linked entry', async () => {
    fetchLinks.mockImplementation(() => Promise.resolve([mockedLinks[0]]));
    const [renderResult] = build();
    await wait();
    expect(renderResult).toMatchSnapshot();
  });

  it('does not render the component if no links are present', async () => {
    fetchLinks.mockImplementation(() => Promise.resolve([]));
    const [renderResult] = build();
    await wait();
    expect(renderResult).toMatchSnapshot();
  });

  it('opens the entry in a slide in editor', async () => {
    fetchLinks.mockImplementation(() => Promise.resolve(mockedLinks));
    const [renderResult] = build();
    await wait();
    fireEvent.mouseOver(renderResult.getByTestId('cf-linked-entities-icon'));
    fireEvent.click(renderResult.getByTestId(`cf-linked-entry-${mockedLinks[0].id}`));
    expect(slideInNavigator.goToSlideInEntity).toHaveBeenCalledWith({
      type: 'Entry',
      id: mockedLinks[0].id
    });
  });
});
