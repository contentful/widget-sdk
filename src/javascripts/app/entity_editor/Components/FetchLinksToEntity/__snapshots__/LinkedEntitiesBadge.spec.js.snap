// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedEntitiesBadge does not render with 1 linked entry 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-1 {
  color: inherit;
}

.emotion-0 {
  margin-right: 2px;
}

.emotion-3 {
  line-height: 0;
}

.emotion-8 {
  padding: 0;
  max-height: 200px;
  overflow-y: auto;
  overflow-x: hidden;
}

.emotion-4 {
  padding: 0 0.5rem;
  background: #e5ebed;
}

.emotion-7 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem;
}

.emotion-7:hover {
  background: #f7f9fa;
}

.emotion-5 {
  margin-right: 1.5rem;
  max-width: 400px;
  color: #2a3039;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 span {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.emotion-6 {
  margin-left: auto;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #8C53C2;
  font-size: 1rem;
  padding: 0px 5px;
  border-radius: 4px;
  color: #ffffff;
  background: #e5ebed;
  color: #536171;
}

<body>
    <div>
      <div
        class="Dropdown__Dropdown___1qyn8 emotion-3"
        data-test-id="cf-ui-dropdown"
        position="bottom-left"
      >
        <div
          aria-expanded="true"
          aria-haspopup="menu"
          class="emotion-2"
          data-test-id="cf-linked-entities-icon"
          title="Used by 1 entry"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD emotion-0"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
          <p
            class="Paragraph__Paragraph___2aO-9 emotion-1"
            data-test-id="cf-ui-paragraph"
          >
            1
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        class="DropdownContainer__DropdownContainer___3WlJM"
        style="top: 0px; left: 0px;"
      >
        <div
          class="InViewport__InViewport___2o6g4"
          data-test-id="cf-ui-in-viewport"
          offset="0"
        >
          <div>
            <div
              class="emotion-4"
              data-test-id="cf-linked-entities-header"
            >
              <p
                class="Paragraph__Paragraph___2aO-9"
                data-test-id="cf-ui-paragraph"
              >
                Used by 1 entry
              </p>
            </div>
            <ul
              class="List__List___1awv4 emotion-8"
              data-test-id="cf-ui-list"
            >
              <li
                class="ListItem__ListItem___3S9Oc emotion-7"
                data-test-id="cf-linked-entry-id1"
              >
                <button
                  class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 emotion-5 TextLink__TextLink--primary___2Vc9F"
                  data-test-id="cf-ui-text-link"
                  type="button"
                >
                  <span
                    class="TabFocusTrap__TabFocusTrap___39Vty"
                    tabindex="-1"
                  >
                    title1
                  </span>
                </button>
                <p
                  class="Paragraph__Paragraph___2aO-9 emotion-6"
                  data-test-id="cf-ui-paragraph"
                >
                  contentTypeName1
                </p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-1 {
  color: inherit;
}

.emotion-0 {
  margin-right: 2px;
}

.emotion-3 {
  line-height: 0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #8C53C2;
  font-size: 1rem;
  padding: 0px 5px;
  border-radius: 4px;
  color: #ffffff;
  background: #e5ebed;
  color: #536171;
}

<div>
    <div
      class="Dropdown__Dropdown___1qyn8 emotion-3"
      data-test-id="cf-ui-dropdown"
      position="bottom-left"
    >
      <div
        aria-expanded="true"
        aria-haspopup="menu"
        class="emotion-2"
        data-test-id="cf-linked-entities-icon"
        title="Used by 1 entry"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD emotion-0"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
          />
        </svg>
        <p
          class="Paragraph__Paragraph___2aO-9 emotion-1"
          data-test-id="cf-ui-paragraph"
        >
          1
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LinkedEntitiesBadge does render the disabled badge component if no links are present 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-1 {
  color: inherit;
}

.emotion-0 {
  margin-right: 2px;
}

.emotion-3 {
  line-height: 0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #8C53C2;
  font-size: 1rem;
  padding: 0px 5px;
  border-radius: 4px;
  color: #ffffff;
  background: #e5ebed;
  color: #536171;
}

<body>
    <div>
      <div
        class="Dropdown__Dropdown___1qyn8 emotion-3"
        data-test-id="cf-ui-dropdown"
        position="bottom-left"
      >
        <div
          aria-expanded="false"
          aria-haspopup="menu"
          class="emotion-2"
          data-test-id="cf-linked-entities-icon"
          title="Reused by 0 entries"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD emotion-0"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
          <p
            class="Paragraph__Paragraph___2aO-9 emotion-1"
            data-test-id="cf-ui-paragraph"
          >
            0
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-1 {
  color: inherit;
}

.emotion-0 {
  margin-right: 2px;
}

.emotion-3 {
  line-height: 0;
}

.emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #8C53C2;
  font-size: 1rem;
  padding: 0px 5px;
  border-radius: 4px;
  color: #ffffff;
  background: #e5ebed;
  color: #536171;
}

<div>
    <div
      class="Dropdown__Dropdown___1qyn8 emotion-3"
      data-test-id="cf-ui-dropdown"
      position="bottom-left"
    >
      <div
        aria-expanded="false"
        aria-haspopup="menu"
        class="emotion-2"
        data-test-id="cf-linked-entities-icon"
        title="Reused by 0 entries"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD emotion-0"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
          />
        </svg>
        <p
          class="Paragraph__Paragraph___2aO-9 emotion-1"
          data-test-id="cf-ui-paragraph"
        >
          0
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`LinkedEntitiesBadge renders the component with 2 linked entries 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #8C53C2;
  font-size: 1rem;
  padding: 0px 5px;
  border-radius: 4px;
  color: #ffffff;
}

.emotion-1 {
  color: inherit;
}

.emotion-0 {
  margin-right: 2px;
}

.emotion-3 {
  line-height: 0;
}

.emotion-11 {
  padding: 0;
  max-height: 200px;
  overflow-y: auto;
  overflow-x: hidden;
}

.emotion-4 {
  padding: 0 0.5rem;
  background: #e5ebed;
}

.emotion-7 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.5rem;
}

.emotion-7:hover {
  background: #f7f9fa;
}

.emotion-5 {
  margin-right: 1.5rem;
  max-width: 400px;
  color: #2a3039;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-5 span {
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.emotion-6 {
  margin-left: auto;
}

<body>
    <div>
      <div
        class="Dropdown__Dropdown___1qyn8 emotion-3"
        data-test-id="cf-ui-dropdown"
        position="bottom-left"
      >
        <div
          aria-expanded="true"
          aria-haspopup="menu"
          class="emotion-2"
          data-test-id="cf-linked-entities-icon"
          title="Reused by 2 entries"
        >
          <svg
            class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--white___3GVPJ emotion-0"
            data-test-id="cf-ui-icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
            />
          </svg>
          <p
            class="Paragraph__Paragraph___2aO-9 emotion-1"
            data-test-id="cf-ui-paragraph"
          >
            2
          </p>
        </div>
      </div>
    </div>
    <div>
      <div
        class="DropdownContainer__DropdownContainer___3WlJM"
        style="top: 0px; left: 0px;"
      >
        <div
          class="InViewport__InViewport___2o6g4"
          data-test-id="cf-ui-in-viewport"
          offset="0"
        >
          <div>
            <div
              class="emotion-4"
              data-test-id="cf-linked-entities-header"
            >
              <p
                class="Paragraph__Paragraph___2aO-9"
                data-test-id="cf-ui-paragraph"
              >
                Reused by 2 entries
              </p>
            </div>
            <ul
              class="List__List___1awv4 emotion-11"
              data-test-id="cf-ui-list"
            >
              <li
                class="ListItem__ListItem___3S9Oc emotion-7"
                data-test-id="cf-linked-entry-id1"
              >
                <button
                  class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 emotion-5 TextLink__TextLink--primary___2Vc9F"
                  data-test-id="cf-ui-text-link"
                  type="button"
                >
                  <span
                    class="TabFocusTrap__TabFocusTrap___39Vty"
                    tabindex="-1"
                  >
                    title1
                  </span>
                </button>
                <p
                  class="Paragraph__Paragraph___2aO-9 emotion-6"
                  data-test-id="cf-ui-paragraph"
                >
                  contentTypeName1
                </p>
              </li>
              <li
                class="ListItem__ListItem___3S9Oc emotion-7"
                data-test-id="cf-linked-entry-id2"
              >
                <button
                  class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 emotion-5 TextLink__TextLink--primary___2Vc9F"
                  data-test-id="cf-ui-text-link"
                  type="button"
                >
                  <span
                    class="TabFocusTrap__TabFocusTrap___39Vty"
                    tabindex="-1"
                  >
                    title2
                  </span>
                </button>
                <p
                  class="Paragraph__Paragraph___2aO-9 emotion-6"
                  data-test-id="cf-ui-paragraph"
                >
                  contentTypeName2
                </p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #8C53C2;
  font-size: 1rem;
  padding: 0px 5px;
  border-radius: 4px;
  color: #ffffff;
}

.emotion-1 {
  color: inherit;
}

.emotion-0 {
  margin-right: 2px;
}

.emotion-3 {
  line-height: 0;
}

<div>
    <div
      class="Dropdown__Dropdown___1qyn8 emotion-3"
      data-test-id="cf-ui-dropdown"
      position="bottom-left"
    >
      <div
        aria-expanded="true"
        aria-haspopup="menu"
        class="emotion-2"
        data-test-id="cf-linked-entities-icon"
        title="Reused by 2 entries"
      >
        <svg
          class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--white___3GVPJ emotion-0"
          data-test-id="cf-ui-icon"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
          />
        </svg>
        <p
          class="Paragraph__Paragraph___2aO-9 emotion-1"
          data-test-id="cf-ui-paragraph"
        >
          2
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
